packages/core/src/components/panel-stack2/panelStack2.tsx

```diff
@@ -97,19 +97,16 @@ export const PanelStack2: PanelStack2Component = <T extends Panel<object>>(props
         stack: propsStack,
     } = props;
     const isControlled = propsStack != null;
-    const [direction, setDirection] = React.useState("push");
 
     const [localStack, setLocalStack] = React.useState<T[]>(initialPanel !== undefined ? [initialPanel] : []);
     const stack = React.useMemo(
         () => (isControlled ? propsStack.slice().reverse() : localStack),
         [localStack, isControlled, propsStack],
     );
     const stackLength = React.useRef<number>(stack.length);
+    // Adjust the direction in case the stack size has changed, controlled or uncontrolled
```

Comment:
nit: this comment doesn't seem to apply anymore now that the state is gone - I think the logic reads pretty explanatory without a comment "if current length is greater than previous length, pop the panel, otherwise push"